### PR TITLE
Add new/missing defmt log format options

### PR DESCRIPTION
**Pull request by bugadani**
Source is https://github.com/probe-rs/probe-rs/blob/08cc1456dd54e8df7a7afe34b0ba0f8345506752/probe-rs/src/bin/probe-rs/util/rtt.rs#L124-L145

This PR depends on a 0.24 release

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -318,6 +326,20 @@
                                                         ],
                                                         "default": "String"
                                                     },
+                                                    "mode": {
+                                                        "type": "string",
```
Comment (bugadani) [inline package.json:339]: I'm just hoping here that omitting this will result in `None` on the Rust side...

**Review by noppej — CHANGES_REQUESTED** [review]
Yes, omitting will result in a `None` on the rust side.

Some observations: 
- It feels odd to have `defmtLogFormat` as a peer to `rttChannelFormats`.  Can we not somehow incorporate it into `rttChannelFormats.dataFormat ... or something sane?
- Why is `channelName` a `number`?

**Comment by bugadani** [comment]
> Why is channelName a number?

Copy-paste-forget bug, sorry

> Can we not somehow incorporate it into

This is a side effect of what cargo embed does: there is an outer option, that serves as a default for unconfigured channels, and inner ones have the same option to override this default. This is probably excessive...

**Comment by noppej** [comment]
@bugadani I just remembered ... we used to have a channel name, and then removed it, because it can be specified in the target application when the channels are configured. Since we would have to arbitrarily choose which one takes priority, it seemed to make more sense to just let the naming happen on the rust side, and then name the client channels based on that.

**Comment by noppej** [comment]
@bugadani I'm not sure I like having the outer and inner option. It adds code complexity and maintenance, for something that the user configures once in a `launch.json` file, and probably very seldom modified after that. 

Does it really add value to have both?

**Comment by bugadani** [comment]
Honestly I don't know, but the option is there in Rust - with the same question attached. I guess we can sack it from `cargo embed`, too, and forget the idea of a default set of configuration altogether.

**Comment by noppej** [comment]
I don't want to propose changing cargo-embed ... it will impact existing users. I am just suggesting that we don't bring the unnecessary complexity to the dap-server. 

Especially since there can only be one defmt channel, what is the point of having an outer setting? Either set it for the one channel you have, or don't. Or am I missing something?

**Comment by bugadani** [comment]
Well then I just added a bunch of code to delete it :)

`cargo embed`'s config format has already been changed in not a small way, so I think twisting it once more shouldn't be too bad. Also, as you've said, this should be a one time annoyance.

File: package.json [hunk 2]
```diff
@@ -318,6 +318,20 @@
...
+                                                        ],
+                                                        "enumDescriptions": [
+                                                            "The target will add data to the channel only if it fits completely, otherwise it will skip the data.",
+                                                            "The target will add as much data to the channel as possible, without blocking.",
+                                                            "The target will block until there is enough space in the channel to add the data."
+                                                        ]
+                                                    },
                                                     "showTimestamps": {
                                                         "type": "boolean",
                                                         "default": false,
```
Comment (noppej) [inline package.json]: @bugadani This default is out of synch with what is implemented on the probe-rs side, where you set default to true. Should we just omit the defaults here?

File: package.json [hunk 3]
```diff
@@ -488,6 +520,10 @@
                                                         "type": "boolean",
                                                         "default": true,
                                                         "description": "Enable the inclusion of Defmt location information in the RTT output for `dataFormat=Defmt`."
+                                                    },
+                                                    "defmtLogFormat": {
```
Comment (noppej) [inline package.json]: @bugadani  The `probe-rs` CLI, uses 
```
      --log-format <LOG_FORMAT>
          The default format string to use for decoding defmt logs
```

Does it make sense for us to be consistent in the naming of these options?

Comment (bugadani) [reply package.json]: Depends on where https://github.com/probe-rs/probe-rs/pull/2442 ends up at, I'm not opposed.

**Comment by bugadani** [comment]
Last commit enables squigglies on typos:

image

Relevant (but not really helpful?) docs: https://json-schema.org/draft/2020-12/json-schema-core#section-10.3.2.3

**Review by noppej — APPROVED** [review]
You are my hero!!! Thank you for always going the extra mile and making things so much better.

**Pull request merged**